android/app/src/main/kotlin/org/getlantern/lantern/model/InAppBilling.kt

```diff
@@ -232,6 +254,24 @@ class InAppBilling(
     private fun handleAcknowledgedPurchases(purchases: List<Purchase>) {
         for (purchase in purchases) {
```

Comment:
You could filter purchases here to only get those with a purchased state that have not yet been acknowledged:

```kotlin
purchases
.filter { it.purchaseState == Purchase.PurchaseState.PURCHASED }
.filterNot { it.isAcknowledged }
```

Reply:
Ah, I am not so sure, If we do this then we lose the purchases that are not in a purchased state, or that are not acknowledged, We have to do consumeAsync regardless of the state, so the user can purchase that again

So if

The user starts Google in-app flow and then goes back without making a purchase, and if we don't do consumeAsync user will see an error 

Let me know what you think.

Reply:
Ah, makes sense then. Thanks for explaining!